### PR TITLE
Switch e2e tests to test tracing with W3C trace context propagation

### PR DESCRIPTION
**Pull request by nareddyt**
GPC's serverless offerings will all start supporting `traceparent` headers soon for integration with OpenTelemetry. go/w3c-trace-context-guidelines

Switch our e2e tests to also use this header for trace context propagation. This will help give a better view of networking latency mentioned in #237. For each compute platform, we can determine exactly how much time is spent between the request leaving ESPv2 and entering the backend.

Here is an example of a cold start that we can now detect because of this PR. My plan is to let this run in our e2e tests for a while, and the evaluate if we should make this the default option for `--tracing_incoming_context` and `--tracing_outgoing_context`. It will make it easier for API Producers to debug cold starts.

image

Signed-off-by: Teju Nareddy <nareddyt@google.com>

### CODE REVIEW

File: tests/e2e/scripts/cloud-run/deploy.sh [hunk 1]
```diff
@@ -324,7 +324,7 @@ function setup() {
     -i "${APIPROXY_IMAGE}"
 
   # Redeploy ESPv2 to update the service config
-  proxy_args="^++^--tracing_sample_rate=0.01++--tracing_outgoing_context=x-cloud-trace-context"
+  proxy_args="^++^--tracing_sample_rate=0.05++--tracing_outgoing_context=traceparent++--tracing_incoming_context=traceparent"
```
Comment (TAOXUY) [inline tests/e2e/scripts/cloud-run/deploy.sh:327]: Why don't we set tracing_incoming_context before?

**Review by TAOXUY — APPROVED** [review]

File: tests/e2e/scripts/cloud-run/deploy.sh (hunk 1 above)
Comment (nareddyt) [reply tests/e2e/scripts/cloud-run/deploy.sh:327]: Technically it's not needed yet. But soon Cloud Run will start publishing traces as well, and we want to keep propagating the trace context.

Comment (nareddyt) [reply tests/e2e/scripts/cloud-run/deploy.sh:327]: This will matter as soon as b/151160669 is fixed. Seems like there is progress maybe made on it.

**Comment by EatonZ** [comment]
@nareddyt Is `x-cloud-trace-context` essentially being deprecated across GCP soon in favor of `traceparent`?

**Comment by nareddyt** [comment]
@EatonZ Recently, there is a unified effort across all GCP serverless products (Cloud Run, Cloud Functions, App Engine) to improve application logging and tracing. They are working on adding new features, such as more traces that display cold start times, networking hops, etc. This will all be backed by OpenTelemetry, which uses the standard `traceparent` header.

Currently our serverless infra is in a transition phrase - it supports both `x-cloud-trace-context` and `traceparent` headers. So you can use either one, they information will be mirrored. But the long-term goal is to switch over to the standard `traceparent` to support distributed tracing across all applications, even your own applications you deploy on GCP. I don't know how long this transition phrase will occur or if they will fully deprecate `x-cloud-trace-context`, but they definitely prefer `traceparent` if possible.

**Pull request merged**